### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by niall-maloney**
### RotaService
Added to replace `RotaHelper`.

**Improvements:**
- Has a dependency on `IClock` injected to allow for deterministic testing.
- Uses NodaTime to simplify or remove much of the date calculations.
- Parameterises const strings bin types for reuse.

### CODE REVIEW

File: src/Sandbox.Core/RotaService.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        {
+            true when !@switch => BinTypes.All,
+            false when @switch => BinTypes.All,
+            true when @switch => BinTypes.Recycling,
+            false when !@switch => BinTypes.Recycling,
+            _ => throw new ArgumentException("Cannot determine which bins."),
+        };
+    }
+
+    public Housemate? GetTurn(IList<Housemate> housemates, int offset)
```
Comment (niall-maloney) [inline src/Sandbox.Core/RotaService.cs:22]: What is `offset`? I can't infer from the name it's purpose.

Comment (niall-maloney) [inline src/Sandbox.Core/RotaService.cs:22]: What is `offset`? I can't infer from the name it's purpose.

File: src/Sandbox.Core.Tests/Sandbox.Core.Tests.csproj [hunk 2]
```diff
@@ -11,7 +11,9 @@
 
     <ItemGroup>
         <PackageReference Include="coverlet.collector" Version="6.0.0"/>
+        <PackageReference Include="FluentAssertions" Version="6.12.0" />
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0"/>
+        <PackageReference Include="NodaTime.Testing" Version="3.1.11" />
```
Comment (niall-maloney) [inline src/Sandbox.Core.Tests/Sandbox.Core.Tests.csproj:16]: https://nodatime.org/3.1.x/userguide/testing

File: src/Sandbox.Core.Tests/Sandbox.Core.Tests.csproj [hunk 3]
```diff
@@ -11,7 +11,9 @@
 
     <ItemGroup>
         <PackageReference Include="coverlet.collector" Version="6.0.0"/>
+        <PackageReference Include="FluentAssertions" Version="6.12.0" />
```
Comment (niall-maloney) [inline src/Sandbox.Core.Tests/Sandbox.Core.Tests.csproj:14]: https://fluentassertions.com/introduction

File: src/Sandbox.Core/Sandbox.Core.csproj [hunk 4]
```diff
@@ -9,6 +9,8 @@
     <ItemGroup>
         <PackageReference Include="Dapper" Version="2.1.35"/>
         <PackageReference Include="FluentAssertions" Version="6.12.0"/>
+        <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0" />
+        <PackageReference Include="NodaTime" Version="3.1.11" />
```
Comment (niall-maloney) [inline src/Sandbox.Core/Sandbox.Core.csproj:13]: https://nodatime.org/3.1.x/userguide/index

File: src/Sandbox.Core/RotaService.cs (hunk 1 above)
Comment (maloneye) [reply src/Sandbox.Core/RotaService.cs:22]: nice, didn't know this!

Comment (maloneye) [reply src/Sandbox.Core/RotaService.cs:22]: Hangover from when this was originally in excel, probably not need now, but its allows you to offset who's turn it is without updating the underling list

File: src/Sandbox.Core/RotaService.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    public string GetBins(bool @switch)
+    {
+        var weekNumber = GetWeekOfWeekYear(GetCurrentDate());
+        return (weekNumber % 2 == 0) switch
+        {
+            true when !@switch => BinTypes.All,
+            false when @switch => BinTypes.All,
+            true when @switch => BinTypes.Recycling,
+            false when !@switch => BinTypes.Recycling,
+            _ => throw new ArgumentException("Cannot determine which bins."),
```
Comment (maloneye) [inline src/Sandbox.Core/RotaService.cs:18]: the default case is impossible to get to here

File: src/Sandbox.Core/RotaService.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+        var weekNumber = GetWeekOfWeekYear(GetCurrentDate());
+        var index = (weekNumber + offset) % housemates.Count;
+        return housemates.ElementAt(index);
+    }
+
+    public LocalDate GetTurnDate(int housematesCount, int rank, int currentRank)
+    {
+        var today = GetCurrentDate();
+        var nextThursday = today.Next(IsoDayOfWeek.Thursday);
```
Comment (maloneye) [inline src/Sandbox.Core/RotaService.cs:37]: Noda time is great, wish i'd found it earlier

File: src/Sandbox.Core/RotaService.cs (hunk 5 above)
Comment (niall-maloney) [reply src/Sandbox.Core/RotaService.cs:18]: It's for completion otherwise static analysis tools warn about it missing a default case.

**Comment by niall-maloney** [comment]
@maloneye do you want to try adding some more unit test cases? Then have a go at plumbing the RotaService in?

**Pull request merged**